File: MinimizeNumber.c
include <stdio.h>

int main() {
    int N;
    scanf("%d", &N);

    int maxOperations = 0;
    int oddPresent = 0;

    for (int i = 0; i < N; i++) {
        int Ai;
        scanf("%d", &Ai);

        if (Ai % 2 != 0) {
            oddPresent = 1;
        }
    }

    if (!oddPresent) {
        int operations = 0;

        while (1) {
            int allEven = 1;

            for (int i = 0; i < N; i++) {
                if (Ai[i] % 2 != 0) {
                    allEven = 0;
                    break;
                }
            }

            if (allEven) {
                operations++;
                for (int i = 0; i < N; i++) {
                    Ai[i] /= 2;
                }
            } else {
                break;
            }
        }

        maxOperations = operations;
    }

    printf("%d\n", maxOperations);

    return 0;
}

File: MinimizeNumber.py
def maximize_operations(N, A):
    max_operations = float('inf')  # Initialize to positive infinity
 
    for i in range(N):
        Ai = A[i]
 
        # Count the number of trailing zeros (even numbers)
        zeros_count = 0
        while Ai % 2 == 0:
            zeros_count += 1
            Ai //= 2
 
        # Update the maximum number of operations
        max_operations = min(max_operations, zeros_count)
 
    return max_operations
 
# Input
N = int(input())
A = list(map(int, input().split()))
 
# Output
result = maximize_operations(N, A)
print(result)
